### DISCUSSION TITLE
Icons

### DISCUSSION THREAD

**Discussion by afcj8** [badges/shields#10588] [Q&A] [answered]
When will Java and C# language icons be incorporated?

**Comment by jNullj** [comment, marked as answer]
As detailed in [this blog post](https://shields.io/blog/sunsetting-shields-custom-logos), icons are derived from the Simple Icons project. If you require a specific icon, you may contribute to the Simple Icons project or use a custom icon encoded in Base64 as a parameter.

**Reply by afcj8** [reply]
How do I contribute and use Base64?

**Comment by chris48s** [comment]
Bit of background on this and https://github.com/badges/shields/discussions/10589

In the past, SimpleIcons used to include icons for a number of Microsoft products. In SimpleIcons 13 these were all removed in response to a request from Microsoft's legal team. It is unlikely that icons for C#, Visual Studio or any other Microsoft product will be included in the SimpleIcons set in future. There is more background on this in https://github.com/simple-icons/simple-icons/pull/10019

Similarly with Java, SimpleIcons used to include an icon for this. Again, this was removed in response to a request from Oracle's legal team and is unlikely to be re-added. More details at https://github.com/simple-icons/simple-icons/issues/7374 Lots of people use The OpenJDK icon instead https://simpleicons.org/?q=openjdk
